src/AccessibilityInsights.SharedUx/Telemetry/TelemetrySink.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using AccessibilityInsights.Extensions;
+using AccessibilityInsights.Extensions.Interfaces.Telemetry;
+using System;
+using System.Collections.Generic;
+
+namespace AccessibilityInsights.SharedUx.Telemetry
+{
+    /// <summary>
+    /// This is the lowest sink for telemetry in Accessibility Insights.
+    /// All telemetry comes through this class.
+    /// It should not be accessed directly, but only through the higher-level telemetry classes
+    /// such as Logger and TelemetryController.
+    /// This class should not throw any exceptions.
+    /// </summary>
+    internal static class TelemetrySink
+    {
+        private static ITelemetry _telemetry => Container.GetDefaultInstance()?.Telemetry;
+
+        /// <summary>
+        /// Whether or not telemetry toggle button is enabled in the settings.
+        /// </summary>
+        public static bool IsTelemetryAllowed { get; internal set; } = false;
+
+        /// <summary>
+        /// Whether or not telemetry is enabled. Exposed to allow callers who do lots of
+        /// work to short-circuit their processing when telemetry is disabled
+        /// </summary>
+        public static bool IsEnabled => IsTelemetryAllowed && _telemetry != null;
+
+        /// <summary>
+        /// Publishes event with single property/value pair to the current telemetry pipeline
+        /// </summary>
+        /// <param name="action"></param>
+        /// <param name="property"></param>
+        /// <param name="value"></param>
+        public static void PublishTelemetryEvent(string action, string property, string value)
+        {
+            PublishTelemetryEvent(action, new Dictionary<string, string>
+            {
+                        { property, value }
+                    });
```

Comment:
indents are off here

Reply:
@RobGallo, ideally lines 41-43 would each have 1 less indent

---
In reply to: [272355080](https://github.com/Microsoft/accessibility-insights-windows/pull/258#discussion_r272355080) [](ancestors = 272355080)